cpp/src/fil/infer.cu

```diff
@@ -282,43 +274,34 @@ struct tree_aggregator_t<NITEMS, TREE_PER_CLASS_MANY_CLASSES> {
         (vec<NITEMS, float>*)((char*)shared_workspace + data_row_size)),
       num_classes(num_classes_) {
     for (int c = threadIdx.x; c < num_classes; c += blockDim.x)
-      per_class_margin[c].fill(0.0f);
-    //__syncthreads(); // done in the main loop
+      per_class_margin[c] = vec<NITEMS, float>();
+    //__syncthreads() is done in infer_k
   }
 
   __device__ __forceinline__ void accumulate(
     vec<NITEMS, float> single_tree_prediction, int tree) {
     // since threads are assigned to consecutive classes, no need for atomics
     per_class_margin[tree % num_classes] += single_tree_prediction;
-    //__syncthreads(); // done in the main loop
+    //__syncthreads() is done in infer_k
```

Comment:
`// __syncthreads() is performed in infer_k` (extra space between `//` and `__syncthreads()`, to avoid confusion with commented-out code).